clippy_lints/src/redundant_type_annotations.rs

```diff
@@ -0,0 +1,201 @@
+use clippy_utils::diagnostics::span_lint;
+use rustc_ast::LitKind;
+use rustc_hir as hir;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::Ty;
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Warns about needless / redundant type annotations.
+    ///
+    /// ### Why is this bad?
+    /// Code without type annotations is shorter and in most cases
+    /// more idiomatic and easier to modify.
+    ///
+    /// ### Example
```

Comment:
It would be good to have a `### Limitations` section, as suggested in this comment:

https://github.com/rust-lang/rust-clippy/pull/10570#issuecomment-1556238326

Reply:
Done thanks!